### PR TITLE
[1.0.0] Slashing Magnitudes

### PR DESCRIPTION
**Pull request by jbrower95**
# Summary
- `podOwnerShares` -> `getWithdrawableShares`
- Updated to use `eigenlayer-contracts` generated ABIs, instead of the ones left in this repo.
- status + find-stale-pods updated.

# status Command
- The shares are computed as `getWithdrawableShares()` over the beacon chain.

# find-stale-pods command
- Rewritten to use multicall.
- Rewritten per our updated computation.

### CODE REVIEW

File: cli/core/status.go [hunk 1]
```diff
@@ -144,11 +146,18 @@ func GetStatus(ctx context.Context, eigenpodAddress string, eth *ethclient.Clien
...
+	PanicOnError("failed to read delegationManager", err)
+
+	delegationManager, err := DelegationManager.NewDelegationManager(delegationManagerAddress, eth)
+	PanicOnError("failed to reach delegationManager", err)
+
+	_, shares, err := delegationManager.GetDepositedShares(nil, eigenPodOwner)
+	PanicOnError("failed to load owner shares", err)
+
+	// TODO: finish computation based on shares/strategies. (this is wrong)
+	currentOwnerSharesETH := IweiToEther(shares[0])
```
Comment (jbrower95) [inline cli/core/status.go:159]: @wadealexc - for the pod status, should we be using getDepositedShares directly in place of podOwnerShares? We probably also need to compute the sum of avs slashings that have occurred, right?

File: cli/core/findStalePods.go [hunk 2]
```diff
@@ -102,7 +104,14 @@ func isEigenpod(eth *ethclient.Client, chainId uint64, eigenpodAddress string) (
...
 
-	podOwnerShares, err := podMan.PodOwnerShares(nil, owner)
+	dmAddress, err := podMan.DelegationManager(nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to reach delegationManager: %w", err)
+	}
+	dm, err := DelegationManager.NewDelegationManager(dmAddress, eth)
+
+	// TODO: are we looking for an aggregate sum here?
+	_, podOwnerShares, err := dm.GetDepositedShares(nil, owner)
```
Comment (jbrower95) [inline cli/core/findStalePods.go:114]: ditto comment here -- we're computing total TVL in `findStalePods` so we can understand whether the pod has slashed by more than N% of its beacon balance

@gpsanant - not sure if the calculation makes sense off deposited shares -- and presumably it's the sum across all strategies

File: cli/core/status.go (hunk 1 above)
Comment (jbrower95) [reply cli/core/status.go:159]: (presumably this also the `sum(shares)` and not `shares[0]` :p )

File: cli/core/status.go [hunk 3]
```diff
@@ -144,11 +147,20 @@ func GetStatus(ctx context.Context, eigenpodAddress string, eth *ethclient.Clien
...
+	delegationManager, err := DelegationManager.NewDelegationManager(delegationManagerAddress, eth)
+	PanicOnError("failed to reach delegationManager", err)
+
+	_, shares, err := delegationManager.GetDepositedShares(nil, eigenPodOwner)
+	PanicOnError("failed to load owner shares", err)
+
+	totalDepositedShares := utils.BigSum(shares)
+
+	// TODO: is this calculation correct?
+	currentOwnerSharesETH := IweiToEther(totalDepositedShares)
```
Comment (jbrower95) [inline cli/core/status.go:162]: cc: @wadealexc @gpsanant

File: cli/core/status.go [hunk 4]
```diff
@@ -144,11 +147,20 @@ func GetStatus(ctx context.Context, eigenpodAddress string, eth *ethclient.Clien
...
-	currentOwnerSharesETH := IweiToEther(currentOwnerShares)
-	currentOwnerSharesWei := currentOwnerShares
+	delegationManagerAddress, err := eigenPodManager.DelegationManager(nil)
+	PanicOnError("failed to read delegationManager", err)
+
+	delegationManager, err := DelegationManager.NewDelegationManager(delegationManagerAddress, eth)
+	PanicOnError("failed to reach delegationManager", err)
+
+	_, shares, err := delegationManager.GetDepositedShares(nil, eigenPodOwner)
+	PanicOnError("failed to load owner shares", err)
```
Comment (wadealexc) [inline cli/core/status.go:157]: The return value here is an array of shares / array of strategies. If you're using this method you only want to sum the shares that correspond to the beacon chain ETH strategy.

That said, I think you want `getWithdrawableShares`, which accepts a strategy as input and will return (withdrawableShares, depositShares)

File: cli/core/findStalePods.go [hunk 5]
```diff
@@ -24,111 +27,188 @@ func PodManagerContracts() map[uint64]string {
...
+			return nil
+		}
+		return mc
+	})
+	if lastError != nil {
+		return nil, lastError
 	}
 
-	expectedPod, err := podMan.OwnerToPod(&bind.CallOpts{}, owner)
+	authoritativeOwnerToPod, err := multicall.DoManyAllowFailures(mc, authoritativeOwnerToPodCalls...)
```
Comment (gpsanant) [inline cli/core/findStalePods.go:118]: none of these should fail

Comment (gpsanant) [reply cli/core/findStalePods.go:118]: they shouldn't be allowed to fail

**Review by seanmcgary — APPROVED** [review]

File: cli/core/findStalePods.go (hunk 5 above)
Comment (jbrower95) [reply cli/core/findStalePods.go]: updated

**Pull request merged**